### PR TITLE
Split generated JsonXExtensions into smaller modular files

### PR DESCRIPTION
**Pull request by DrakslerT**
## Summary

The annotation processor previously generated all extension functions and serialization infrastructure into a single monolithic `JsonXExtensions.kt` file. This refactors the generator to emit three focused files while preserving the same public API surface via `@file:JvmMultifileClass`.

**Related issue**: None

### Type

- [ ] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [x] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

### Description

All three generated files carry `@file:JvmName("JsonXExtensions")` + `@file:JvmMultifileClass`, so the API surface is unchanged from a consumer perspective.

**Generated file breakdown:**

- **`{Model}Extensions.kt`** (one per annotated class) — model-specific extension functions:
  - `Original.toResourceObject(meta, links)`, `Model.toResourceObject()`, `Item.toResourceObject()`
  - `Model.toJsonApiX()`, `List.toJsonApiXList()`
  - `Model.toJsonApiXString(...)`, `List.toJsonApiXString(...)`

- **`JsonXSerializerModule.kt`** — serialization infrastructure shared across all models:
  - `jsonApiXSerializerModule` (polymorphic registrations for all models, custom links/errors/meta)
  - `format` (`Json` instance)

- **`JsonXCoreExtensions.kt`** — shared stateless utilities:
  - `toManyRelationshipModel()` / `toOneRelationshipModel()`
  - `decodeJsonApiError<T>()`
  - `asJsonXHttpException<T>()` (emitted only when the retrofit module is on the classpath)
  - `decodeJsonApiXString()` / `decodeJsonApiXListString()`

**Processor changes:**

- `JsonXExtensionsSpecGenerator` now collects each holder's data into a `ClassInfo` map and delegates to the three new builders.
- `JsonXCoreExtensionsSpecBuilder`, `JsonXModelExtensionsSpecBuilder`, `JsonXSerializerModuleSpecBuilder` — new builder objects replacing the old monolithic `JsonXExtensionsSpecBuilder`.
- `JsonApiConstants.FileNames` — added `JSON_X_CORE_EXTENSIONS` and `JSON_X_SERIALIZER_MODULE` constants.

## Checklist

- [x] I have performed a self-review of my own code.
- [ ] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [ ] I have updated the documentation to reflect my changes (if applicable).

## Additional notes

Compile-time output grows from 1 file to `N + 2` files (where N = number of annotated models), but incremental compilation benefit improves since only the affected model's file needs regeneration on change.

### CODE REVIEW

**Comment by DrakslerT** [comment]
@copilot  generate a summary of the changes

File: processor/src/main/java/com/infinum/jsonapix/processor/specs/jsonxextensions/JsonXCoreExtensionsSpecBuilder.kt [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+        fileSpec.addAnnotation(
+            AnnotationSpec.builder(JvmName::class)
+                .addMember("%S", JsonApiConstants.FileNames.JSON_X_EXTENSIONS)
+                .useSiteTarget(AnnotationSpec.UseSiteTarget.FILE).build(),
+        )
+        fileSpec.addAnnotation(
+            AnnotationSpec.builder(JvmMultifileClass::class)
+                .useSiteTarget(AnnotationSpec.UseSiteTarget.FILE).build(),
+        )
```
Comment (DrakslerT) [inline processor/src/main/java/com/infinum/jsonapix/processor/specs/jsonxextensions/JsonXCoreExtensionsSpecBuilder.kt:49]: This is added as before, the JsonXExtensions file used the JVM name:
Image

Now all the files have:
Image

Not sure if its even needed though @thisAAY 🤔

**Comment by DrakslerT** [comment]
@thisAAY the current structure is like this: 
Screenshot 2026-04-08 at 10 31 11

Do we want to change it in any way? or spread it
